src/Validation/Traits/ImageValidationTrait.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Josegonzalez\Upload\Validation\Traits;
+
+trait ImageValidationTrait
+{
+    /**
+     * Check that the file is above the minimum width requirement
+     *
+     * @param mixed $check Value to check
+     * @param int $width Width of Image
+     * @return bool Success
+     */
+    public static function isAboveMinWidth($check, $width)
+    {
+        // Non-file uploads also mean the height is too big
+        if (!isset($check['tmp_name']) || !strlen($check['tmp_name'])) {
+            return false;
```

Comment:
Conditional validation can be done through a callback on the 'on' parameter: http://book.cakephp.org/3.0/en/core-libraries/validation.html#conditional-validation